File: examples/python/basic.py
"""
Menstrual Cycle Calculator API - Basic Usage Example

This example demonstrates the basic usage of the Menstrual Cycle Calculator API.
API Documentation: https://docs.apiverve.com/ref/menstrualcycle
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/menstrualcycle'

def call_menstrualcycle_api():
    """
    Make a GET request to the Menstrual Cycle Calculator API
    """
    try:
        # Query parameters
        params &#x3D; {&#x27;last_period&#x27;: &#x27;2024-01-01&#x27;, &#x27;cycle_length&#x27;: 28, &#x27;period_length&#x27;: 5, &#x27;cycles&#x27;: 3}

        headers = {
            'x-api-key': API_KEY
        }

        response = requests.get(API_URL, headers=headers, params=params)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Menstrual Cycle Calculator API...\n')

    result = call_menstrualcycle_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: npm/examples/basic.js
/**
 * Basic Example - Menstrual Cycle Calculator API
 *
 * This example demonstrates how to use the Menstrual Cycle Calculator API.
 * Make sure to set your API key in the .env file or replace '[YOUR_API_KEY]' below.
 */

require('dotenv').config();
const menstrualcycleAPI = require('../index.js');

// Initialize the API client
const api = new menstrualcycleAPI({
    api_key: process.env.API_KEY || '[YOUR_API_KEY]'
});

// Example query
var query = {
  last_period: "2024-01-01",
  cycle_length: 28,
  period_length: 5,
  cycles: 3
};

// Make the API request using callback
console.log('Making request to Menstrual Cycle Calculator API...\n');

api.execute(query, function (error, data) {
    if (error) {
        console.error('Error occurred:');
        if (error.error) {
            console.error('Message:', error.error);
            console.error('Status:', error.status);
        } else {
            console.error(JSON.stringify(error, null, 2));
        }
        return;
    }

    console.log('Response:');
    console.log(JSON.stringify(data, null, 2));
});
